plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -399,7 +442,15 @@ public Optional<List<Object>> getTableExecuteSplitsInfo()
             return Optional.empty();
         }
         requireNonNull(scannedFiles, "scannedFiles is null");
+        long filesSkipped = 0;
+        if (scannedFilesByPartition != null) {
+            filesSkipped = scannedFilesByPartition.values().stream()
+                    .filter(Optional::isPresent)
+                    .count();
+            scannedFilesByPartition = null;
+        }
         List<Object> splitsInfo = ImmutableList.copyOf(scannedFiles.build());
+        log.info("Generated %d splits, skipped %d files for OPTIMIZE", splitsInfo.size(), filesSkipped);
```

Comment:
Why not debug?

Reply:
I think it's useful enough to be at info, until we enhance optimize to print some useful summary at the end of execution. We get a bunch of useless logs out of iceberg library anyway in IcebergSplitSource, this isn't adding much to the noise.